packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -196,6 +197,10 @@ export class AuthUtil {
         return this.conn !== undefined && isBuilderIdConnection(this.conn)
     }
 
+    public isInternalAmazonUser(): boolean {
+        return this.isConnected() && this.startUrl === 'https://amzn.awsapps.com/start'
```

Comment:
nitL Was this ok to expose startUrl to public?

Reply:
Yes. There has been a lot of back and forth on this (on Slack) and, ultimately, the decision is that this is okay.
 https://github.com/aws/aws-toolkit-jetbrains/pull/4869#discussion_r1750938610

Reply:
Can we move this utility function to a common place that can be use by Amazon Q and Toolkits.